src/github.com/matrix-org/go-neb/services/google/google.go

```diff
@@ -0,0 +1,221 @@
+// Package google implements a Service which adds !commands for Google custom search engine.
+// Initially this package just supports image search but could be expanded to provide other functionality provided by the Google custom search engine API - https://developers.google.com/custom-search/json-api/v1/overview
+package google
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"net/http"
+	"net/url"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/matrix-org/go-neb/types"
+	"github.com/matrix-org/gomatrix"
+)
+
+// ServiceType of the Google service
+const ServiceType = "google"
+
+var httpClient = &http.Client{}
+
+// Unsused -- leaving this in place for the time being to show structure of the request
+// type googleQuery struct {
+// 	// Query search text
+// 	Query string `json:"q"`
+// 	// Number of search results
+// 	Num int `json:"num"`
+// 	// Search result offset
+// 	Start int `json:"start"`
+// 	// Size of images to serch for (usually set to "medium")
+// 	ImgSize string `json:"imgSize"`
+// 	// Type of search - Currently always set to "image"
+// 	SearchType string `json:"searchType"`
+// 	// Type of image file to retur64 `json:"totalResults"`
+// 	FileType string `json:"fileType"`
+// 	// API key
+// 	Key string `json:"key"`
+// 	// Custom serch engine ID
+// 	Cx string `json:"cx"`
+// }
+
+type googleSearchResults struct {
+	SearchInformation struct {
+		TotalResults int64 `json:"totalResults,string"`
```

Comment:
Without the string annotation the json lib returns an error (json: cannot unmarshal string into Go value of type int64).

As mentioned in the json.Marshall docs (https://golang.org/pkg/encoding/json/#Marshal) -- "The "string" option signals that a field is stored as JSON inside a JSON-encoded string. It applies only to fields of string, floating point, integer, or boolean types. This extra level of encoding is sometimes used when communicating with JavaScript programs:", which appears to fix the issue.